mongorestore/restore.go

```diff
@@ -331,11 +332,29 @@ func (restore *MongoRestore) RestoreIntent(intent *intents.Intent) Result {
 	return result
 }
 
-func stripInvalidOptions(indexes []IndexDocument) []IndexDocument {
+func convertLegacyIndexes(indexes []IndexDocument) []IndexDocument {
 	for i, index := range indexes {
+		for j, elem := range index.Key {
+			switch elem.Value {
+			case true:
+				md := bsonutil.MarshalD(indexes[i].Key)
+				indexJSON, err := md.MarshalJSON()
+				if err != nil {
+					indexJSON = []byte("<error formatting index>")
+				}
+				indexes[i].Key[j].Value = 1
+				log.Logvf(log.Always, "convertLegacyIndexes: converting the legacy vaule of '%s' from '%s' to '%s' in index '%s'", indexes[i].Key[j].Key, "true", "1", string(indexJSON))
+			}
+		}
+
 		for key := range index.Options {
-			if _, ok := indexOptions[key]; !ok {
+			if _, ok := validIndexOptions[key]; !ok {
+				optionsJSON, err := bson.MarshalExtJSON(indexes[i].Options, true, false)
+				if err != nil {
+					optionsJSON = []byte("<error formatting index>")
+				}
 				delete(indexes[i].Options, key)
+				log.Logvf(log.Always, "convertLegacyIndexes: removing invalid option '%v' from index options '%v'", key, string(optionsJSON))
```

Comment:
Same.  I suggest saving all the logging for after all options are updated.  Rather than saying "removed this option", you can just say "updated options: Old: <old>; New: <new>".